x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/index.tsx

```diff
@@ -332,6 +339,14 @@ export const AgentListPage: React.FunctionComponent<{}> = () => {
     }, {} as { [k: string]: AgentPolicy });
   }, [agentPolicies]);
 
+  const isAgentSelectable = (agent: Agent) => {
+    if (!agent.active) return false;
+
+    const agentPolicy = agentPolicies.find((p) => p.id === agent.policy_id);
```

Comment:
I am wondering if we should index policy per id at some point we are going to loop a lot of time over this policy list

Reply:
actually looks like we do `agentPoliciesIndexedById`

Reply:
Good eyes. Thanks! This replaced a few instances of .find with byId but looks likes it missed one (some?). I'll review and fix as a part of the next PR in this series.

Reply:
Fixed in https://github.com/elastic/kibana/commit/18fca79d69ad83637e25c2292bdf5f0867b2b446